### PR TITLE
fix(engine,client): re-anchor life-magnitude citations and tighten shortcut guards

### PR DESCRIPTION
**Pull request by lgray**
## Summary

Review residue from the loop-shortcut interaction preview that merged as `4d5d34bd7`: citation
subjects re-anchored to the rule they actually state, stale doc comments corrected, two wildcard
matches spelled exhaustively so the compiler gates a future variant, boundary siblings for the
payload guard's refusals, a modal panel that fails closed instead of rendering an empty control,
test fixtures typed instead of cast past their brands, and the wire-version guard bound to the role
each numeral names. No runtime behaviour changes outside the modal's fail-closed arm.

## Files changed

- `crates/engine/src/game/interaction.rs` — citation subjects; doc comment; two exhaustive matches
- `crates/engine/src/game/visibility.rs` — citation subjects
- `crates/engine/src/types/game_state.rs` — citation subjects
- `crates/engine/src/types/interaction.rs` — citation subjects
- `crates/engine/tests/integration/loop_shortcut.rs` — citation subjects
- `crates/engine/tests/integration/interaction_contract.rs` — citation subjects
- `crates/engine/tests/integration/fantastic_four_bounded_loop.rs` — citation subjects
- `crates/server-core/src/interaction_payload_guard.rs` — three accept-at-the-ceiling siblings
- `client/src/components/modal/LoopShortcutModal.tsx` — fail closed with nothing to state
- `client/src/components/modal/__tests__/LoopShortcutModal.test.tsx` — brand-typed fixtures
- `client/src/adapter/__tests__/ws-adapter.test.ts` — brand-typed fixtures
- `client/src/adapter/__tests__/server-draft-adapter.test.ts` — brand-typed fixtures
- `client/src/adapter/generated/interaction/index.ts` — regenerated bindings
- `scripts/check-protocol-version.mjs` — bind each version to its role; scope the refusal to titles

## Track

Developer

## LLM

Model: claude-opus-5
Tier: Frontier
Thinking: high

## Implementation method (required)

Method: /engine-implementer

## CR references

`CR 704.5a` (kept where the statement is a seat reaching 0 and being removed), `CR 119.3`
(re-anchored where the statement is the size of a life delta or the seat it is charged to),
`CR 732.2a`, `CR 732.2b`, `CR 704`. Every number was verified by grepping `docs/MagicCompRules.txt`
with the rule's SUBJECT checked against the statement it heads.

## Verification

- [x] Required checks ran clean, or the exact CI-owned alternative is stated below.
- [x] Gate A output below is for the current committed head.
- [x] Final review-impl below is clean for the current committed head.
- [x] Both anchors cite existing analogous code at the same seam.

All commands below ran at the committed head `4e5f5ffffd7deffaa85272c4b5e8621a043c6c88`.

- `cargo fmt --all -- --check` — exit 0
- `cargo clippy --workspace --all-targets -- -D warnings` — exit 0
- `cargo nextest run --workspace` — exit 0; 30485 tests run, 30485 passed, 45 skipped
- `./scripts/check-interaction-bindings.sh --check` — exit 0
- `node scripts/check-protocol-version.mjs` — exit 0
- `./scripts/check-parser-combinators.sh` — exit 0 (Gate A, below)
- `pnpm install --frozen-lockfile` — exit 0
- `pnpm exec tsc -b --noEmit --force` — exit 0
- `pnpm lint` — exit 0
- `pnpm test` — exit 1; 5807 passed, 1 failed, 12 todo. The single failure is
  `client/src/audio/__tests__/AudioManagerDisabled.test.ts`, which this PR does not reach and did
  not introduce:
  - `git diff --stat 755b35edf..4e5f5ffff -- client/src/audio/` is empty; the same command without
    the pathspec is non-empty.
  - The file passes 23/23 when run alone.
  - Two full-suite runs at this head failed *different* members of that file in opposite directions.
  - The identical `pnpm test` at the branch point `755b35edf` — none of these commits present —
    fails the same file (5805 passed, 1 failed).
  - CI's own `Frontend (lint, type-check, test)` job is green on `main` at `726553573`, so this is a
    local-parallelism artifact of the base tree rather than a CI-visible failure.

## Gate A

Gate A PASS head=4e5f5ffffd7deffaa85272c4b5e8621a043c6c88 base=d6d28370c09242182488cac72e16e5a84941885f

## Anchored on

- crates/engine/src/game/interaction.rs:6454 — `loop_shortcut_choices` already matches
  `LoopShortcutCandidateValue` exhaustively, one named arm per variant and no wildcard; the two
  matches this PR converts are spelled to that same shape.
- crates/server-core/src/interaction_payload_guard.rs:122 —
  `rejects_an_oversized_nested_shortcut_pin_budget` is the existing refusal test whose size is
  derived from the bound rather than written as a literal; the accept-at-the-ceiling siblings this
  PR adds sit beside it and derive their sizes the same way.

## Final review-impl

Final review-impl PASS head=4e5f5ffffd7deffaa85272c4b5e8621a043c6c88

## Claimed parse impact

None.

## Scope Expansion

None.

## Validation Failures

None.

## CI Failures

None.

## Summary by CodeRabbit

- **Bug Fixes**
  - Prevented shortcut declarations when no valid choice has been selected.
  - Improved shortcut response rendering so declaration details appear only when applicable.

- **Documentation**
  - Updated rules references and explanatory text related to shortcut previews, victim charges, visibility, and announcements.

- **Tests**
  - Added coverage for valid boundary values in interaction payload validation.
  - Expanded shortcut modal and preview tests for valid and invalid answer scenarios.

### CODE REVIEW

**Review by ntindle — APPROVED** [review]
**Approved** — current head `4e5f5ffffd7deffaa85272c4b5e8621a043c6c88`.

Review residue from the loop-shortcut preview: life-magnitude comments correctly re-anchored from CR 704.5a (SBA) to CR 119.3 (life adjustment); CR 704.5a kept where elimination bounds apply. Exhaustive `LoopShortcutCandidateValue` matches, modal fail-closed when may wording cannot be stated, payload-guard boundary siblings, and tighter protocol-version title scoping. No PROTOCOL/WIRE bump required.

File: scripts/check-protocol-version.mjs [hunk 1]
```diff
@@ -322,10 +322,20 @@ requirePattern(serverCoreSource, new RegExp(`fn protocol_version_is_${P}(?![0-9]
...
-  "client/src/network/__tests__/protocol.test.ts");
+// Both legs read the file's test TITLES, not its whole source, so coverage that legitimately
+// drives the superseded version in a body is not a bump leftover. Ceiling: double-quoted titles
+// only, so a backtick or single-quoted title falls out of the slice — admit-only, never a false
+// refusal, which is what makes the narrowing safe.
+const p2pProtocolTestTitles = [
+  ...p2pProtocolTestSource.matchAll(/\b(?:describe|it|test)\(\s*"([^"]*)"/g),
+]
+  .map((match) => match[1])
+  .join("\n");
```
[the remark this answers is not in the document]
Comment (lgray) [reply scripts/check-protocol-version.mjs:333]: Confirmed for the gate-block legs, and one part of this is worse than the comment above them admits — but it is tracked as a follow-up rather than fixed on this head, which is already approved.

The two `requirePattern` calls at 361-366 do scan the raw slice, and a gate block whose executable test has been deleted but whose comment still quotes it satisfies **both** legs. Probed against the two regexes with controls: comment-only block -> title leg true, frames leg true; real test -> both true; neither present -> both false. The line above them discloses that exposure for the `it(...)` title leg only; the `setupFrameAt` leg has the identical exposure and is not disclosed. That gap is the real defect here, and it is booked.

One correction to the premise: the `matchAll` at 330 is not a raw scan for the check itself. It extracts titles first and both legs at 336-338 run over the extracted list, not over the file. The residue there is narrower than stated - the extractor is a regex, so a commented-out `it("...")` is harvested as if it were a title - but a `setupFrameAt(43)` in a body or a comment cannot reach those two legs at all.

Parsing test call expressions is the right shape for the fix. It is a change to a check script with its own failure modes in both directions (a stricter extractor can also red a correct tree), so it goes in its own change against main rather than onto an approved head whose review and gate records name this exact commit.
